util/options.py

```diff
@@ -1,31 +1,186 @@
 import json
+import logging
 import os
+import subprocess
 
 import yaml
+from pydantic import BaseModel, SecretStr, constr
+from pydantic_settings import BaseSettings
 
+logger = logging.getLogger(__name__)
 
-class Options:
-    def __init__(self):
-        super(Options, self).__init__
+def BitwardenConfig(settings_dict: dict):
+    '''
+    Takes a dict of settings loaded from yaml and adds the secrets from bitwarden to the settings dict.
+    The bitwarden secrets are mapped to the settings dict using the bitwarden_mapping dict.
+    The secrets are sourced based on a project id in the settings dict.
+    '''
+    logger.debug("Loading secrets from Bitwarden")
+    try:
+        command = f"bws secret list {settings_dict['bws']['project_id']}"
+        bitwarden_raw = subprocess.check_output(command, shell=True, text=True)
+    except Exception as e:
+        logger.error(e)
+    bitwarden_settings = parse_json_to_dict(bitwarden_raw)
 
-    def fetch(path="config/options.yml"):
-        # Get file path
-        full_path = os.path.join(os.getcwd(), path)
+    bitwarden_mapping = {
+        'discord_bot_token': ('discord', 'bot_token'),
+        'discord_client_id': ('discord', 'client_id'),
+        'discord_secret': ('discord', 'secret'),
+        'stripe_api_key': ('stripe', 'api_key'),
+        'stripe_webhook_secret': ('stripe', 'webhook_secret'),
+        'stripe_price_id': ('stripe', 'price_id'),
+        'email_password': ('email', 'password'),
+        'jwt_secret': ('jwt', 'secret'),
+        'infra_wifi': ('infra', 'wifi'),
+        'infra_application_credential_id': ('infra', 'application_credential_id'),
+        'infra_configuration_credential_secret': ('infra', 'application_credential_secret')
+    }
 
-        # Load options.
-        with open(full_path, "r") as file:
-            options = yaml.safe_load(file)
+    bitwarden_mapped = {}
+    for bw_key, nested_keys in bitwarden_mapping.items():
+        if bw_key in bitwarden_settings:
+            top_key, nested_key = nested_keys
+            if top_key not in bitwarden_mapped:
+                bitwarden_mapped[top_key] = {}
+            bitwarden_mapped[top_key][nested_key] = bitwarden_settings[bw_key]
 
-        return options
+    for top_key, nested_dict in bitwarden_mapped.items():
+        if top_key in settings_dict:
+            for nested_key, value in nested_dict.items():
+                settings_dict[top_key][nested_key] = value
+    return settings_dict
 
-    def get(self, arg=None):
-        options = self.fetch()
+settings_dict = dict()
 
-        return options.get(arg, None)
+# Reads config from ../config/options.yml
+here = os.path.abspath(os.path.dirname(__file__))
```

Comment:
Instead of doing this raw in the python file, it's probably best to make `Settings()` into a different class and add this to the constructor with "options.yml" as the default value.

Reply:
I am going to be honest I have no idea how that function works, chatgpt wrote it